server/project/project.go

```diff
@@ -405,8 +405,7 @@ func (s *Server) GetSyncWindowsState(ctx context.Context, q *project.SyncWindows
 }
 
 func (s *Server) NormalizeProjs() {
-	ctx := context.Background()
-	projList, err := s.List(ctx, &project.ProjectQuery{})
+	projList, err := s.appclientset.ArgoprojV1alpha1().AppProjects(s.ns).List(metav1.ListOptions{})
```

Comment:
I think `NormalizeProjs` should return error if normalization fails after retries and server should crash. Overwise we won't noticed that upgrade is not smooth.